config/container_runtime_executors.go

```diff
@@ -0,0 +1,34 @@
+package config
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type ContainerRuntimeExecutors []ContainerRuntimeExecutor
+
+func (e ContainerRuntimeExecutors) Select(labels labels.Labels) (string, error) {
+	for _, c := range e {
+		ok, err := c.Matches(labels)
+		if err != nil {
+			return "", err
+		}
+		if ok {
+			return c.Name, nil
+		}
+	}
+	return "", nil
+}
+
+type ContainerRuntimeExecutor struct {
+	Name     string               `json:"name"`
```

Comment:
the `Name` is kind of confusing.  It should be `executor` or` executorName` or `executorType`

Reply:
I think that "type" and "name" are synonymous. I.e. there is only one executor of type "pns", it is the "pns" executor.

I'm not sure about the correct term here, but I'd like to leave this called "name" unless you feel strongly?